lib/db/helpers/index.js

```diff
@@ -27,7 +31,206 @@ let findOne = (profileID) => {
   })
 }
 
+let findRoomByName = (allrooms, title) => {
+  //findIndex is the new es6 method
+  let findRoom = allrooms.findIndex((element, index, array) => {
+    if (element.title === title) {
+      return true;
+    } else {
+      return false;
+    }
+  });
+
+  return findRoom > -1 ? true : false;
+}
+
+// a function that generates a unique roomID
+let randomHex = () => {
+  return crypto.randomBytes(24).toString('hex');
+}
+
+
+let getAllRooms = (callback) => {
+  var listName = 'rooms';
+  redisClient.hgetall(listName, (err, response) => {
+    if (err) {
+      /* istanbul ignore next */
+      return callback(err);
+    } else {
+      //convert object into arr
+      let arr = Object.keys(response).map((key) => {
+        return response[key];
+      });
+      //parsed each object
+      let parsed = arr.map((obj) => {
+        return JSON.parse(obj);
+      });
+
+      return callback(null, parsed)
+    }
+  });
+}
+
+let createNewRoom = (newRoomInput, callback) => {
+
+  let roomObject = {};
+  roomObject.title = newRoomInput;
+  roomObject.id = randomHex();
+  roomObject.timestamp = Date.now();
+  roomObject.users = [];
+
+  let listName = 'rooms';
+
+  redisClient.hmset(listName, newRoomInput, JSON.stringify(roomObject), (error, response) => {
+    if (error) {
+      /* istanbul ignore next */
+      return callback(error)
+    } else {
+      return callback(null, response);
+    }
+  });
+}
+
+//refactor this function
+let getRoom = (title, callback) => {
+  var listName = 'rooms';
+  redisClient.hmget(listName, title, (err, response) => {
+    if (err) {
+      /* istanbul ignore next */
+      return callback(err);
+    } else {
+
+      let parsedObj = response.map((object, index) => {
+          return JSON.parse(object);
+      });
+
+      let found = parsedObj.filter((room) => {return room.title === title});
+
+      return callback(null, found[0])
+    }
+  });
+}
+
+//add users to chatrooms
+let addUserToRoom = (data, token, socket, callback) => {
+  //get the room object
+  let getCurrentRoom = getRoom(data.roomTitle, (errR, room) => {
+    if (room !== undefined) {
+      //get the active user's id ( id as used in session )
+      let userId = getUserId(token, (err, user) => {
+        let userParsed = JSON.parse(user);
+        //check if this user already exist in chatroom
+
+          let checkUser = room.users.findIndex((element, index, array) => {
```

Comment:
This could be refactored to a one-liner if you return element.userId === userParsed.id

```js
let checkUser = room.users.findIndex((element, index, array) => element.userId === userParsed.id);
```